src/functionality.py

```diff
@@ -49,24 +142,722 @@ def _validate_command(ctx: commands.Context) -> bool:
+            if suffix:
+                final_message += f'{suffix}'
+
+            await message.delete()
+            await to_channel.send(final_message)
+
+        # Checks Admin Status
+        def admin_check(ctx):
+            admin_roles = [role for role in ctx.guild.roles if role.permissions.administrator]
+
+            for admin_role in admin_roles:
+                if admin_role in ctx.author.roles:
+                    return True
+
+            return None
+
+        # Converts Seconds to Days
+        def second_to_day(second: int):
+            answer = second / 86400
+            return round(answer)
+
+        # Converts Seconds to Hours
+        def second_to_hour(second: int):
+            answer = second / 3600
+            return round(answer)
+
+        def ban_profile_check(_id):
+            ban_list = ban_id_check()
+            found = []
+
+            for user in ban_list:
+                if str(_id) == user["discord_id"]:
+                    found.append(user)
+
+            return found
+
+        def ban_profile_check_all():
+
+            ban_list = ban_id_check()
+            found = []
+
+            for ban in ban_list:
+                found.append(ban)
+
+            return found
+
+        def update_unban(_id):
+            ban_ids = ban_id_check()
+            updated_ban_list = list()
+            inform_ban_list = list()  # This is to inform players when their ban is over.
+
+            for user in ban_ids:
+                if str(_id) == user["discord_id"]:
+                    update_dict = {"discord_id": user["discord_id"], "discord_name": user["discord_name"], "ban_timestamp": user["ban_timestamp"],
+                                   "ban_length": user["ban_length"], "reason": user["reason"], "ended": 1}
+
+                    inform_ban_list.append(update_dict)
+
+                else:
+                    update_dict = user
+
+                updated_ban_list.append(update_dict)
+
+            with open("ban_ids.csv", "w+") as file:
+                fieldnames = ["discord_id", "discord_name", "ban_timestamp", "ban_length", "reason", "ended"]
+                writer = csv.DictWriter(file, fieldnames=fieldnames)
+                writer.writeheader()
+
+                for update in updated_ban_list:
+                    writer.writerow(update)
+
+            return updated_ban_list, inform_ban_list
+
+        def update_ban_list():
+            ban_ids = ban_id_check()
+            updated_ban_list = list()
+            inform_ban_list = list()  # This is to inform players when their ban is over.
+            update_dict = None
+
+            for user in ban_ids:
+                answer = int(user["ban_timestamp"]) + int(user["ban_length"])
+                answer = round(time.time()) - answer
+
+                if answer >= 0:
+                    if bool(int(user["ended"])) is True:
+                        update_dict = user
+
+                    if bool(int(user["ended"])) is False:
```

Comment:
Use if user["ended"] != "1" or similar. There should be no Python that code that uses is True or is False.